### ISSUE TITLE
`DraggableScrollableSheet` programmatically dragged doesn't maintain size after rebuild

### ISSUE THREAD

**Issue by lionelmennig** [flutter/flutter#114342]
## Steps to Reproduce

1. Run code below
2. Try expand the sheet by scroll, tap light and dark button
3. Minimize sheet, now expand it by tapping the button, tap light and dark button

**Expected results:** Sheet maintain it size

**Actual results:** Sheet lost the size and went back to initial size during rebuild if previous size was changed from controller

<details>
<summary>Code sample</summary>

```
import 'package:flutter/material.dart';

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();

  runApp(const App());
}

class App extends StatefulWidget {
  const App({Key? key}) : super(key: key);

  @override
  State<App> createState() => AppState();

  static AppState of(BuildContext context) {
    return context.findAncestorStateOfType<AppState>()!;
  }
}

class AppState extends State<App> {
  ThemeMode _themeMode = ThemeMode.light;

  void setThemeMode(ThemeMode mode) {
    if (_themeMode != mode) {
      setState(() {
        _themeMode = mode;
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      themeMode: _themeMode,
      theme: ThemeData(brightness: Brightness.light),
      darkTheme: ThemeData(brightness: Brightness.dark),
      home: const Home(),
    );
  }
}

class Home extends StatelessWidget {
  const Home({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Demo'),
      ),
      body: Center(
        child: ElevatedButton(
          onPressed: () {
            showModalBottomSheet(
              context: context,
              isScrollControlled: true,
              builder: (context) => const _DemoSheet(),
            );
          },
          child: const Text('Open'),
        ),
      ),
    );
  }
}

class _DemoSheet extends StatefulWidget {
  const _DemoSheet({Key? key}) : super(key: key);

  @override
  _DemoSheetState createState() => _DemoSheetState();
}

class _DemoSheetState extends State<_DemoSheet> {
  final _draggableController = DraggableScrollableController();

  void _expand() {
    _draggableController.animateTo(
      0.9,
      duration: const Duration(milliseconds: 150),
      curve: Curves.easeOut,
    );
  }

  void _setTheme(ThemeMode themeMode) {
    App.of(context).setThemeMode(themeMode);
  }

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return DraggableScrollableSheet(
      controller: _draggableController,
      initialChildSize: 0.5,
      minChildSize: 0.2,
      maxChildSize: 0.9,
      expand: false,
      builder: (context, scrollController) => ListView(
        physics: const BouncingScrollPhysics(),
        controller: scrollController,
        children: [
          Padding(
            padding: const EdgeInsets.all(8.0),
            child: ElevatedButton(
              onPressed: _expand,
              child: const Text('Expand'),
            ),
          ),
          Padding(
            padding: const EdgeInsets.all(8.0),
            child: Text(
              'Current brightness: ${theme.brightness}',
              textAlign: TextAlign.center,
            ),
          ),
          Padding(
            padding: const EdgeInsets.all(8.0),
            child: ElevatedButton(
              onPressed: () => _setTheme(ThemeMode.light),
              child: const Text('Light'),
            ),
          ),
          Padding(
            padding: const EdgeInsets.all(8.0),
            child: ElevatedButton(
              onPressed: () => _setTheme(ThemeMode.dark),
              child: const Text('Dark'),
            ),
          ),
          for (int i = 0; i < 100; i++)
            Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                i.toString(),
                textAlign: TextAlign.center,
              ),
            ),
        ],
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel stable, 3.3.6, on macOS 12.5 21G72 darwin-x64, locale en-GB)
    • Flutter version 3.3.6 on channel stable at /Users/*/Flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6928314d50 (6 days ago), 2022-10-25 16:34:41 -0400
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0

[!] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/*/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13F100
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • iPad Pro (9.7-inch) (mobile) • 3B8FD730-4BA9-444C-866A-D6BB7616D597 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    • macOS (desktop)              • macos                                • darwin-x64     • macOS 12.5 21G72 darwin-x64
    • Chrome (web)                 • chrome                               • web-javascript • Google Chrome 106.0.5249.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

Basically, I duplicated the issue #101114 since it's still occurring the same exact way (I just did put my `doctor output` in place of). Initial author: @SuperPenguin 
Initial fixer: @caseycrogers

**Comment by xu-baolin** [comment]
@lionelmennig Hi, thanks for reporting.
Can you test the master channel branch, the patch has not been released to the stable branch yet.

**Comment by lionelmennig** [comment]
> @lionelmennig Hi, thanks for reporting. Can you test the master channel branch, the patch has not been released to the stable branch yet.

Hi! You're right, it works on `master`! Thanks!
